pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -30,30 +30,33 @@ import (
 
 const primaryPodInterfaceName = "eth0"
 
-type VMNetworkConfigurator struct {
+type NetworkingConfigurator interface {
+	Setup() error
+}
+
+type InfraNetworkingConfigurator struct {
 	vmi          *v1.VirtualMachineInstance
 	handler      netdriver.NetworkHandler
 	cacheFactory cache.InterfaceCacheFactory
+	launcherPID  int
+	nicMaker     NicGenerator
 }
 
-func newVMNetworkConfiguratorWithHandlerAndCache(vmi *v1.VirtualMachineInstance, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return &VMNetworkConfigurator{
-		vmi:          vmi,
-		handler:      handler,
-		cacheFactory: cacheFactory,
-	}
-}
-
-func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &netdriver.NetworkUtilsHandler{}, cacheFactory)
+type NetworkingSpecGenerator struct {
+	vmi          *v1.VirtualMachineInstance
+	handler      netdriver.NetworkHandler
+	cacheFactory cache.InterfaceCacheFactory
+	domXML       api.Domain
+	nicMaker     NicGenerator
 }
```

Comment:
What do you think about embedding the struct instead of having a member ? this we we don't need a "nicMaker" field

```golang
type NetworkingSpecGenerator struct {
        NicGenerator
	vmi          *v1.VirtualMachineInstance
	handler      netdriver.NetworkHandler
	cacheFactory cache.InterfaceCacheFactory
	domXML       api.Domain	
}
```

constructor example

https://play.golang.org/p/Tg2Gk4QCn7J